java/fury-core/src/main/java/org/apache/fury/resolver/ClassInfo.java

```diff
@@ -87,13 +86,15 @@ public class ClassInfo {
       this.fullClassNameBytes = null;
     }
     if (cls != null
-        && (classId == ClassResolver.NO_CLASS_ID || classId == ClassResolver.REPLACE_STUB_ID)) {
+        && ((classId == ClassResolver.NO_CLASS_ID
+                && !classResolver.getFury().getConfig().isMetaShareEnabled())
+            || classId == ClassResolver.REPLACE_STUB_ID)) {
```

Comment:
In addition, there is another question, that is, why do we need to create `packageNameBytes` and `classNameBytes` when classId is `ClassResolver.REPLACE_STUB_ID`? I see `packageNameBytes` and `classNameBytes` are only used when `classInfo.classId == NO_CLASS_ID && !metaContextShareEnabled`

Reply:
Sometimes only classes are serialized, not the instance. If we serialize such class only, we need to write classname bytes. Could you document it?